File: Python/12. Integer to Roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        num_copy = num
        roman = [
            'I', 'V', 'X', 'L', 'C', 'D', 'M'
        ]
        indeks = 0
        result = ''

        while indeks <= 6:
            mod = num_copy % 10
            num_copy //= 10
            if (mod == 0):
                pass
            elif (mod <= 3):
                result = roman[indeks] * mod + result
            elif (mod == 4):
                result = roman[indeks] + roman[indeks + 1] + result
            elif (mod == 9):
                result = roman[indeks] + roman[indeks + 2] + result
            else:
                result = roman[indeks + 1] + roman[indeks] * (mod - 5) + result
            if num_copy == 0:
                break
            indeks += 2
            print(num_copy)

        return result

File: Javascript/12. Integer to Roman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
    let numCopy = num
    let roman = [
        'I', 'V', 'X', 'L', 'C', 'D', 'M'
    ]
    let index = 0
    let result = ''

    while (index <= 6) {
        let mod = numCopy % 10
        numCopy = Math.floor(numCopy/10)

        if (mod == 0) {
        } else if (mod <= 3) {
            result = roman[index].repeat(mod) + result
        } else if (mod == 4) {
            result = (roman[index] + roman[index + 1]) + result
        } else if (mod == 9) {
            result = (roman[index] + roman[index + 2]) + result
        } else {
            result = (roman[index + 1] + roman[index].repeat(mod-5)) + result
        }
        index += 2
    }
    return result
};
